src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2570,6 +2577,35 @@ PruneShardList(Oid relationId, Index tableId, List *whereClauseList,
 
 
 /*
+ * ContainsFalseClause returns whether the flattened where clause list
+ * contains false as a clause.
+ */
+static bool
```

Comment:
I am about to add the exact same function to router planner. Perhaps we should put this into a more visible location for reuse purposes.